syntax/go.vim

```diff
@@ -162,17 +162,17 @@ endif
 syn match       goSingleDecl        /\%(import\|var\|const\) [^(]\@=/ contains=goImport,goVar,goConst
 
 " Integers
-syn match       goDecimalInt        "\<-\=\(0\|[1-9]_\?\(\d\+_\?\)*\)\%([Ee][-+]\=\d\+\)\=\>"
-syn match       goDecimalErrorInt   "\<-\=\(_\(\d\+_*\)\+\|\([1-9]\d*_*\)\+_\{2,\}\(\d\+_*\)\+\|\([1-9]\d*_*\)\+_\+\)\%([Ee][-+]\=\d\+\)\=\>"
+syn match       goDecimalInt        "\<-\=\(0\|[1-9]_\?\(\d\|\d\+_\?\d\+\)*\)\%([Ee][-+]\=\d\+\)\=\>"
+syn match       goDecimalError      "\<-\=\(_\(\d\+_*\)\+\|\([1-9]\d*_*\)\+__\(\d\+_*\)\+\|\([1-9]\d*_*\)\+_\+\)\%([Ee][-+]\=\d\+\)\=\>"
 syn match       goHexadecimalInt    "\<-\=0[xX]_\?\(\x\+_\?\)\+\>"
-syn match       goHexadecimalError  "\<-\=0[xX]_\?\(\x\+_\?\)*\(\([^ \t0-9A-Fa-f_]\|_\{2,\}\)\S*\|_\)\>"
+syn match       goHexadecimalError  "\<-\=0[xX]_\?\(\x\+_\?\)*\(\([^ \t0-9A-Fa-f_]\|__\)\S*\|_\)\>"
 syn match       goOctalInt          "\<-\=0[oO]\?_\?\(\o\+_\?\)\+\>"
 syn match       goOctalError        "\<-\=0[0-7oO_]*\(\([^ \t0-7oOxX_/\]\}]\|[oO]\{2,\}\|__\)\S*\|_\|[oOxX]\)\>"
```

Comment:
Does `goOctalError` need to include `bB` in the classes that also check for `oOxX`?

Reply:
As I mentioned above, it depends on how we want to make it works. Technically, it works correctly without `bB`. Binary rules placed below so overrides octal errors if needed. But to make it fully independent should be expanded to a couple of other symbols like `.`, `i` that currently overrides by floats and imaginary rules. What's your point of view on it?

Reply:
Fair point. 

Let's just add a test to cover the case in case these ever get over around

Sorry for asking again: it can be difficult to keep track of everything in these long running PRs

Reply:
Yeah, will do! That's fine, I also started to confuse in addressed comments.